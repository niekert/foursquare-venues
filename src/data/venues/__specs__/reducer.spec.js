import reduce from '../reducer';
import {
  FETCH_VENUES_BY_GEO,
  FETCH_VENUES_BY_SEARCH,
  FETCH_VENUES_SUCCESS,
  FETCH_VENUES_ERROR,
  CLOSE_SELECTED_VENUE,
  SELECT_VENUE,
} from '../actions';

describe('venuesReducer', () => {
  test(`${FETCH_VENUES_BY_GEO}`, () => {
    const initialState = { mock: 'true ' };
    const result = reduce(initialState, {
      type: FETCH_VENUES_BY_GEO,
      payload: { latLng: 'mock' },
    });

    expect(result).toMatchSnapshot();
  });

  test(`${FETCH_VENUES_BY_SEARCH}`, () => {
    const result = reduce(
      {},
      {
        type: FETCH_VENUES_BY_SEARCH,
        payload: { query: 'mock_query ' },
      },
    );

    expect(result).toMatchSnapshot();
  });

  test(`${FETCH_VENUES_SUCCESS}`, () => {
    const initialState = {
      isLoading: true,
      mock_key: 'mock',
    };

    const result = reduce(initialState, {
      type: FETCH_VENUES_SUCCESS,
      payload: {
        venues: 'mock_venues',
      },
    });

    expect(result).toMatchSnapshot();
  });

  test(`${SELECT_VENUE}`, () => {
    const intialState = {
      mock_key: 'do_not_overwrite',
    };

    const result = reduce(intialState, {
      type: SELECT_VENUE,
      payload: {
        venueId: 'selected_mock_venue',
      },
    });

    expect(result).toMatchSnapshot();
  });
});
