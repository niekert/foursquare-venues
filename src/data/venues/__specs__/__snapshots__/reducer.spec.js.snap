// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`venuesReducer FETCH_VENUES_BY_GEO 1`] = `
Object {
  "currentGeoLatLng": "mock",
  "isLoading": true,
  "mock": "true ",
  "searchType": "search_current_geo",
}
`;

exports[`venuesReducer FETCH_VENUES_BY_SEARCH 1`] = `
Object {
  "isLoading": true,
  "query": "mock_query ",
  "searchType": "search_query",
}
`;

exports[`venuesReducer FETCH_VENUES_SUCCESS 1`] = `
Object {
  "isLoading": false,
  "mock_key": "mock",
  "venues": "mock_venues",
}
`;

exports[`venuesReducer SELECT_VENUE 1`] = `
Object {
  "mock_key": "do_not_overwrite",
  "selectedVenueId": "selected_mock_venue",
}
`;
